linera-views/src/views/mod.rs

```diff
@@ -154,6 +154,10 @@ pub enum ViewError {
     /// The value is too large for the client
     #[error("The value is too large for the client")]
     TooLargeValue,
+
+    /// Blob not found when trying to read it.
+    #[error("Blob not found on storage read: {0}")]
+    BlobNotFoundOnRead(BlobId),
```

Comment:
In the long run we should probably treat all `NotFound` errors the same: either split them up into one variant for each type of value, or keep them all the same variant.

(But is this needed in the first place? Aren't we already handling this at the call sites?)

Reply:
(Yes, it's just weird that `linera-views` and `linera-storage` are treating this particular "not found" error different from the others, just because another crate happens to use it in that way. Maybe we should have a variant for each of them in the long run?)

Reply:
I also feel like this, and the `get_blobs_not_found` function, are symptoms of us carrying errors through too many layers of abstraction.

At the storage layer we don't know whether we were expecting to always find this blob or whether we now need to request it or whatever. That's not what storage is about.
But the caller knows, so it should be the caller that turns this error into the appropriate error, depending on the context.

(Doesn't have to be in this PR, of course.)